content/simulator/programming/time.md

```diff
@@ -0,0 +1,40 @@
+---
+title: Time
+---
+
+In the simulated environment, time advances only at the pace that the simulator is run and in cooperation with the controller code (such as the code controlling your robot). As a result, using `time.time` to know how long your robot has been running for or `time.sleep` to wait for some duration will have unpredictable results, especially when run on a computer other than the one you are developing on (such as during the competition).
+
+Instead your code should explicitly signal to the simulator when it is ready for time to advance and should query for the current time from the simulator. This can be done by using the methods detailed below.
+
+{{% notice tip %}}
+If you find that the simulator freezes then this indicates that your code is busy doing something without giving the simulator a chance to run. This usually indicates that you have a loop somewhere which is expecting time to advance on its own and which should be modified to call `R.sleep` occasionally (even a very small value will allow the simulator to progress).
+{{% /notice %}}
```

Comment:
need for example snippet?

Reply:
Putting code in notices doesn't really work, and even then I'm not sure what this snippet would look like?